src/controllers/sendMail.js

```diff
@@ -0,0 +1,21 @@
+import sgMail from '@sendgrid/mail';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+sgMail.setApiKey(process.env.SENDGRID_APIKEY);
+
+// function to send emai
+let url;
+const sendingMail = (emailAddres) => {
+  url = `http://localhost:3000/user/confirmation/${emailAddres}`;
```

Comment:
Hi @Theoneste1  you have hardcoded
1. the m port: this means your codes will run only in your cmomputer
2. the host (localhost): this means your codes will not run on Heroku. 

you need to use environment variables please.

Reply:
Done with the feedbacks,

1. added the variables in .env
2. return an object instead of message.